pkg/fqdn/re/re.go

```diff
@@ -38,7 +38,9 @@ func CompileRegex(p string) (*regexp.Regexp, error) {
 	r, ok := lru.Get(p)
 	lru.Unlock()
 	if ok {
-		return r.(*regexp.Regexp), nil
+		if reRet, ok := r.(*regexp.Regexp); ok {
+			return reRet, nil
+		}
```

Comment:
Should we return an error otherwise?

Reply:
First of all, I'm pretty sure this is impossible. Assuming it isn't, moving on to compilation seems perfectly acceptable.